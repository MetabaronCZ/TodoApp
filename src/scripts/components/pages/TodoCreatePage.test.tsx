import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { TodoCreatePage } from 'components/pages/TodoCreatePage';

import { mockStore } from 'test/store';
import { withMockedProviders } from 'test/component';

describe('components/pages/TodoCreatePage', () => {
  it('should render correctly', () => {
    const store = mockStore();

    const { container } = render(
      withMockedProviders(
        <Provider store={store}>
          <TodoCreatePage />
        </Provider>,
      ),
    );
    expect(container).toMatchSnapshot();
  });
});
