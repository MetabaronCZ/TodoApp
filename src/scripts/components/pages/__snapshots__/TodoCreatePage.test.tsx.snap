// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/TodoCreatePage should render correctly 1`] = `
.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 10px 0;
}

.c7 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c28 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

.c29 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-right: 5px;
  background: #f0f0f0;
}

.c29:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c29:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c29:disabled {
  color: #c0c0c0;
}

.c36 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
}

.c36:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c36:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c36:disabled {
  color: #c0c0c0;
}

.c6 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  padding-right: 5px;
  background: #f0f0f0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c6:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c6:disabled {
  color: #c0c0c0;
}

.c13 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c13:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c13:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c13:disabled {
  color: #c0c0c0;
}

.c15 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  padding: 0 10px;
  font-size: 20px;
  line-height: 40px;
  font-weight: bold;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

.c38 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #22c;
}

.c38:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c38:focus {
  outline: 1px dotted #22c;
  outline-offset: -1px;
}

.c37 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  padding: 20px 0;
  text-align: center;
}

.c0 {
  min-height: 100vh;
  width: 100%;
  padding: 0 20px;
  background: #f0f0f0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 320px;
  max-width: 980px;
  min-height: 100vh;
  width: 100%;
  margin: 0 auto;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 20px;
}

.c11 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  border-bottom: 1px solid #c0c0c0;
}

.c12:last-child {
  border-bottom: none;
}

.c14 {
  width: 100%;
  background: #c0c0c0;
  text-align: left;
}

.c14:hover {
  background-color: #c0c0c0;
}

.c16 {
  width: 100%;
  text-align: left;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 20px;
}

.c10 {
  width: 240px;
}

.c18 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
}

.c19 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 18px;
  line-height: 30px;
  font-weight: normal;
  margin: 0;
}

.c23 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #222;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: 30px;
}

.c31 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  width: 20px;
  height: 20px;
  border: 2px solid #222;
  color: #222;
  background: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c31:hover {
  background-color: #fcfcfc;
}

.c31:focus-within {
  outline: 1px dotted #222;
}

.c32 {
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 0;
  opacity: 0;
  outline: none;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c26 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  border: none;
  outline: none;
}

.c26:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c26:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c26:disabled {
  color: #c0c0c0;
}

.c34 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  border: none;
  outline: none;
  min-height: 110px;
  resize: vertical;
}

.c34:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c34:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c34:disabled {
  color: #c0c0c0;
}

.c27 {
  position: relative;
  display: inline-block;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 10px;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: block;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 10px;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  min-height: 30px;
}

.c24 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c20 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: calc(30px + 1px);
  border-bottom: 1px solid #c0c0c0;
  background: #f0f0f0;
}

.c35 {
  border-top: 1px solid #c0c0c0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          ☑ 
          TODOLIST
        </a>
        <div
          class="c4"
        />
        <div
          class="c5"
        >
          <a
            class="c6"
            href="/folder/list"
            title="Složky"
          >
            <i
              class="c7"
            >
              ★
            </i>
          </a>
          <a
            class="c6"
            href="/settings"
            title="Nastavení"
          >
            <i
              class="c7"
            >
              ✎
            </i>
          </a>
        </div>
      </header>
      <main
        class="c8"
      >
        <div
          class="c9"
        >
          <nav
            class="c10"
          >
            <ul
              class="c11"
            >
              <li
                class="c12"
              >
                <a
                  class="c13 c14"
                  href="/?folder="
                >
                  <i
                    class="c7"
                  >
                    ★
                  </i>
                  <span
                    class="c15"
                  >
                    Všechny úkoly
                  </span>
                </a>
              </li>
              <li
                class="c12"
              >
                <a
                  class="c13 c16"
                  href="/folder/list"
                >
                  <i
                    class="c7"
                  >
                    ✎
                  </i>
                  <span
                    class="c15"
                  >
                    Složky
                  </span>
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="c17 c18"
          >
            <div>
              <h1
                class="c19"
              >
                Nový úkol
              </h1>
              <div
                class="c20"
              >
                <div>
                  <a
                    class="c13"
                    href="/"
                  >
                    <i
                      class="c7"
                    >
                      ❮
                    </i>
                    <span
                      class="c15"
                    >
                      Zpět
                    </span>
                  </a>
                </div>
              </div>
              <ul
                class="c11"
              >
                <li
                  class="c12"
                >
                  <div
                    class="c21"
                  >
                    <div
                      class="c22"
                    >
                      <label
                        class="c23"
                        for="form-field-0"
                      >
                        Název úkolu:
                      </label>
                    </div>
                    <div
                      class="c24"
                    >
                      <div
                        class="c25"
                      >
                        <input
                          class="c26"
                          id="form-field-0"
                          maxlength="80"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  class="c12"
                >
                  <div
                    class="c21"
                  >
                    <div
                      class="c22"
                    >
                      <label
                        class="c23"
                        for="form-field-1"
                      >
                        Složka:
                      </label>
                    </div>
                    <div
                      class="c24"
                    >
                      <div
                        class="c27"
                      >
                        <button
                          class="c28 c29"
                          id="form-field-1"
                          type="button"
                        >
                          <span
                            class="c15"
                          >
                            -Vyberte složku-
                          </span>
                          <i
                            class="c7"
                          >
                            ˅
                          </i>
                        </button>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  class="c12"
                >
                  <div
                    class="c21"
                  >
                    <div
                      class="c22"
                    >
                      <label
                        class="c23"
                        for="form-field-2"
                      >
                        Hotovo:
                      </label>
                    </div>
                    <div
                      class="c24"
                    >
                      <div
                        class="c30"
                      >
                        <label
                          class="c31"
                          for="form-field-2"
                        >
                          <input
                            class="c32"
                            id="form-field-2"
                            type="checkbox"
                          />
                        </label>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  class="c12"
                >
                  <div
                    class="c33"
                  >
                    <div
                      class="c22"
                    >
                      <label
                        class="c23"
                        for="form-field-3"
                      >
                        Popis úkolu:
                      </label>
                    </div>
                    <div
                      class="c24"
                    >
                      <div
                        class="c25"
                      >
                        <textarea
                          class="c34"
                          id="form-field-3"
                          maxlength="500"
                        />
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
              <div
                class="c35"
              >
                <button
                  class="c28 c36"
                  type="button"
                >
                  <i
                    class="c7"
                  >
                    +
                  </i>
                  <span
                    class="c15"
                  >
                    Vytvořit
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </main>
      <footer
        class="c37"
      >
        <div
          class=""
        >
          Copyright © 2023 MetabaronCZ (
          <a
            class="c38"
            href="https://github.com/MetabaronCZ"
            target="_blank"
          >
            https://github.com/MetabaronCZ
          </a>
          ). All Rights Reserved.
        </div>
      </footer>
    </div>
  </div>
</div>
`;
